module-audio/tags_fetcher/TagsFetcher.cpp

```diff
@@ -20,7 +20,12 @@ namespace tags::fetcher
 
         std::optional<Tags> fetchTagsInternal(const std::string &filePath)
         {
+#ifdef PRODUCT_BellHybrid
+            std::size_t maxTagSize = 200 * 1024;
+            const TagLib::ConstMemoryConstrainedFileRef tagReader(filePath.c_str(), maxTagSize);
+#else
             const TagLib::ConstFileRef tagReader(filePath.c_str());
+#endif
```

Comment:
This is not ideal as we are still injecting product-specific logic to the common part(but this time on the different layer). Unfortunately, I don't see how we can achieve such a functionality without reworking bigger part of system.

Reply:
We could use `ConstMemoryConstrainedFileRef` for Pure as well and pass `numeric_limits<size_t>::max()`. This way there would be no limit for id3v2 tag max size but for flac files pictures would still be ignored.

Reply:
done